src/iterators/chunked_read_ahead_iterator.rs

```diff
@@ -0,0 +1,442 @@
+use std::any::Any;
+use std::panic::{catch_unwind, resume_unwind, AssertUnwindSafe};
+use std::sync::mpsc::{sync_channel, Receiver};
+use std::thread::{self, JoinHandle};
+use std::vec::IntoIter;
+
+// type aliased to get clippy to not think this is too complex
+type PanicUnwindErr = Box<dyn Any + Send>;
+
+/// Iterator extension that spawns an additional thread to read-ahead in the iterator. Sends
+/// results back to this object via a channel and returns them in the same manner as a normal
+/// iterator. This is useful in the context in which the reading of an iterator (or iterators) is
+/// time consuming (e.g. reading from a compressed FASTQ file) and the main thread is bottlenecked
+/// by the speed of the underlying iterator.
+///
+/// To use on a struct that implements ``IntoIter``, it is as simple as:
```

Comment:
I'm confused by all the references to `IntoIter`.  a) `IntoIter` is a struct returned by `Vec::into_iter()`, not a trait (I think you mean `IntoIterator`), but in general there is no requirement to implement that.  You just need an iterator whose Item type is `Send + 'static`.